fcrepo-kernel/src/main/java/org/fcrepo/kernel/observer/FedoraEvent.java

```diff
@@ -59,6 +65,7 @@
      */
     public FedoraEvent(final Event e) {
         checkArgument(e != null, "null cannot support a FedoraEvent!");
+        eventID = pidMinter.mintPid();
```

Comment:
@awoods ' idea will work, but only if we can guarantee that the only place that `FedoraEvent`s will be produced will be implementations of `InternalExternalEventMapper` (including, but _not_ limited to, `AllNodeEventsOneEvent`). Otherwise we need an actual factory. I hate Spring. In CDI, this problem just wouldn't arise.